clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -753,7 +756,8 @@ public static AbstractResponse parseResponse(ByteBuffer responseBuffer, RequestH
     private void processDisconnection(List<ClientResponse> responses,
```

Comment:
I'm thinking, instead of passing boolean flags around, would it be better to have another function to explicitly calling out `processTimeoutDisconnection` ?

Reply:
Done.